cf-key/cf-key-functions.c

```diff
@@ -232,19 +232,13 @@ int RemoveKeys(const char *input, bool must_be_coherent)
 
 static bool KeepKeyPromisesRSA(RSA *pair, const char *public_key_file, const char *private_key_file)
 {
-    int fd = safe_open_create_perms(private_key_file, O_WRONLY | O_CREAT | O_TRUNC, CF_PERMS_DEFAULT);
-
-    if (fd < 0)
-    {
-        Log(LOG_LEVEL_ERR, "Couldn't open private key file '%s' (open: %s)", private_key_file, GetErrorStr());
-        return false;
-    }
-
-    FILE *fp = fdopen(fd, "w");
+    FILE *fp = safe_fopen_create_perms(private_key_file, "w",
+                                       CF_PERMS_DEFAULT);
```

Comment:
As @vpodzime pointed out, private keys should not use the default perms macro, even though they are currently corrrect. Either make a separate macro `CF_PERMS_PRIVKEY`, or keep the hard coded values.